eventuals/task.h

```diff
@@ -241,12 +244,11 @@ struct _TaskFromToWith final {
 
     template <typename Error>
     void Fail(Error&& error) {
-      std::exception_ptr exception;
-
-      exception = make_exception_ptr_or_forward(
-          std::forward<Error>(error));
-
-      Dispatch(Action::Fail, std::nullopt, std::move(exception));
+      if constexpr (tuple_contains_exact_type_v<Error, Catches_>) {
```

Comment:
```suggestion
      // NOTE: we only propagate an upstream error into our type-erased eventual
      // if the eventual catches the error, otherwise we skip it all together.
      if constexpr (tuple_contains_exact_type_v<Error, Catches_>) {
```